snagsby/sources.py

```diff
@@ -63,7 +56,50 @@ def options(self):
 
     @property
     def region_name(self):
-        return self.options.get('region')
+        return self.options.get('region') or 'us-west-2'
+
+    def get_data(self):
+        return sanitize(self.get_raw_data())
+
+class SMSource(AWSSource):
+    def get_sm_response(self):
+        key = "{}{}".format(self.url.netloc, self.url.path)
+        session = boto3.session.Session()
+        endpoint_url = "https://secretsmanager.{}.amazonaws.com".format(self.region_name)
+        client = session.client(
+            service_name='secretsmanager',
+            region_name=self.region_name,
+            endpoint_url=endpoint_url
+        )
+
+        try:
+            return client.get_secret_value( SecretId=key )
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                print("The requested secret " + key + " was not found")
+            elif e.response['Error']['Code'] == 'InvalidRequestException':
+                print("The request was invalid due to:", e)
+            elif e.response['Error']['Code'] == 'InvalidParameterException':
+                print("The request had invalid params:", e)
+            return {}
+
+    def get_raw_data(self):
+        response = self.get_sm_response()
+        if 'SecretString' in response:
+            try:
+                return json.loads(response['SecretString'])
+            except json.decoder.JSONDecodeError as e:
+                return {}
```

Comment:
It could be good to send some debugging logging text if the SecretString isn't there, to help devs understand if they're expecting something back and they get nothing.